### ISSUE TITLE
valid_docs rule shouldn't be enforced for common UIKit method overrides

### ISSUE THREAD

**Issue by litso** [realm/SwiftLint#284]
For example if I override shouldAutoroate I shouldn't have to document the return type.

```
        override func shouldAutorotate() -> Bool {
            return false
        }
```

**Comment by keith** [comment]
I brought this up somewhere but IMO we should ignore everything that is an `override`

**Comment by proth** [comment]
I think so too

**Comment by jpsim** [comment]
The `missing_docs` rule actually skips all overridden declarations, but the `valid_docs` rule validates all documentation-style comments (`///` and `/** */`).

So what's the motivating factor for writing documentation-style comments for an overridden declaration without those docs being fully valid? If it's a note on the implementation to developers (e.g. not users of the API), then why not use a regular style comment? If it is for users of the API, the docs should be complete, no?

**Comment by Sega-Zero** [comment]
I have slightly different case with the same warning. Having this protocol and implementation:
```swift
/// test proto description
public protocol TestProto {
    /// - Returns: some string
    /// - Parameter sampleParam: yet another parameter
    func sampleFunc(sampleParam: String) -> String
}

/// Sample proto documentation
public struct SampleStruct: TestProto {
    /// - seealso: `TestProto`
    public func sampleFunc(sampleParam: String) -> String {
        return ""
    }
}
```

I use [jazzy](https://github.com/realm/jazzy) for doc generation, `SampleStruct` is the part of my public API and should be documented. Why do I get a warning on `SampleStruct.sampleFunc(_:)` implementation?

**Comment by jpsim** [comment]
> Why do I get a warning on SampleStruct.sampleFunc(_:) implementation?

Because it is only partially documented.

**Comment by Sega-Zero** [comment]
Ok, what should be added to make it go away?
![experiments swift 2015-12-25 18-19-17](https://cloud.githubusercontent.com/assets/2203199/12003510/05da4e06-ab34-11e5-9d70-3205935244ac.jpg)

**Comment by Sega-Zero** [comment]
I don't mind to duplicate the description from the protocol, but that's as weird as documenting the method override, i think.

**Comment by norio-nomura** [comment]
> Ok, what should be added to make it go away?

@Sega-Zero Current valid_docs rules implementation is case sensitive and expecting lower case keywords such as `parameter` or `returns`.
It seems that Xcode ignores case of them, so we need to fixing SwiftLint.

**Comment by pcantrell** [comment]
Note that Swiftlint insisting on `parameter` and `returns` sections to match the method signature is contrary to [Apple’s API design guidelines](https://swift.org/documentation/api-design-guidelines.html#special-instructions):

<img width="701" alt="screen shot 2015-12-28 at 10 39 33 am" src="https://cloud.githubusercontent.com/assets/378162/12021815/77d6f302-ad4f-11e5-8fc1-7178c4ee36fe.png">

(This is under “Make documentation comments tool-friendly.”)

**Comment by jpsim** [comment]
Thanks for sharing that link @pcantrell. I've been struggling with this since originally writing the rule, where it's not always appropriate to follow the Swift documentation syntax *exactly*. Of course, it's always possible to just disable a rule, either locally or globally, but I'm open to loosening the constraints on the valid docs rule to reduce the number of false violations.

**Comment by pcantrell** [comment]
It’s the inherent problem of linting tools: the rules that are truly absolute tend to be fairly mindless. Most fall into a nuanced area that requires some human judgement, and deciding where to draw the linter line is a tricky call!

**Comment by marcelofabri** [comment]
Closing this since we removed this rule.
